homeassistant/components/renault/renault_entities.py

```diff
@@ -0,0 +1,131 @@
+"""Base classes for Renault entities."""
+from __future__ import annotations
+
+from typing import Any, cast
+
+from renault_api.kamereon.enums import ChargeState, PlugState
+from renault_api.kamereon.models import (
+    KamereonVehicleBatteryStatusData,
+    KamereonVehicleChargeModeData,
+    KamereonVehicleCockpitData,
+    KamereonVehicleHvacStatusData,
+)
+
+from homeassistant.helpers.entity import DeviceInfo, Entity
+from homeassistant.helpers.update_coordinator import CoordinatorEntity
+from homeassistant.util import slugify
+
+from .renault_vehicle import RenaultVehicleProxy
+
+ATTR_LAST_UPDATE = "last_update"
+
+
+class RenaultDataEntity(CoordinatorEntity, Entity):
+    """Implementation of a Renault entity with a data coordinator."""
+
+    def __init__(
+        self, vehicle: RenaultVehicleProxy, entity_type: str, coordinator_key: str
+    ) -> None:
+        """Initialise entity."""
+        super().__init__(vehicle.coordinators[coordinator_key])
+        self.vehicle = vehicle
+        self._entity_type = entity_type
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return a device description for device registry."""
+        return self.vehicle.device_info
+
+    @property
+    def unique_id(self) -> str:
+        """Return a unique identifier for this entity."""
+        return slugify(f"{self.vehicle.details.vin}-{self._entity_type}")
+
+    @property
+    def name(self) -> str:
+        """Return the name of this entity."""
+        return f"{self.vehicle.details.vin}-{self._entity_type}"
```

Comment:
I would suggest on dropping the VIN from the entity names. It probably won't look friendly and since there is already device support, the entities would be grouped in a device

Reply:
I wasn't sure about this one.
I was worried about users who have more than one vehicle, and by default they would simply have two "Battery Level" sensors but I do agree that VIN isn't the more user friendly. Could/Should I use the model name instead (Zoe/Captur/Twingo...)?

Reply:
Implemented